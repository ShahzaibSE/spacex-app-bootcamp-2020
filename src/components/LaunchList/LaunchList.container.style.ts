import { makeStyles, Theme, createStyles } from '@material-ui/core/styles';

export const launchlist_container_error_message = makeStyles((theme: Theme) => ({
    root: {
      width: '100%',
      '& > * + *': {
        marginTop: theme.spacing(2),
      },
    },
}));

export const backdrop_loader_styles = makeStyles((theme: Theme) =>
      createStyles({
        backdrop: {
          zIndex: theme.zIndex.drawer + 1,
          color: '#fff',
        },
      }),
);