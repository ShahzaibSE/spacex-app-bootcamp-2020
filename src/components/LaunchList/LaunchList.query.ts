import gql from "graphql-tag"

export const query_launch_list = gql `
    query launches{
        launches {
            id
            details
            launch_date_local
            launch_date_unix
            launch_date_utc
            launch_site {
                site_id
                site_name
                site_name_long
            }
            launch_success
            launch_year
            links {
                article_link
                video_link
                flickr_images
            }
            mission_id
            mission_name
        }
    } 
`