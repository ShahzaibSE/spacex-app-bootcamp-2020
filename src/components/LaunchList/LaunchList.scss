.launch_list_container {
    // margin-top:20px;
    // margin-left:10%;
    // margin-right:10%;
}