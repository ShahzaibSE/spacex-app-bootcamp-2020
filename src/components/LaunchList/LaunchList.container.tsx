import React, {useState} from 'react'
import {useLaunchesQuery} from "../../generated/graphql"
import Grid from "@material-ui/core/Grid"
import LinearProgress from '@material-ui/core/LinearProgress'
import MuiAlert, { AlertProps } from '@material-ui/lab/Alert';
import Snackbar from '@material-ui/core/Snackbar';
// Launch list component.
import LaunchList from "./LaunchList"
// Assets.
import {launchlist_container_error_message} from "./LaunchList.container.style"

function Alert(props: AlertProps) {
    return <MuiAlert elevation={6} variant="filled" {...props} />;
}

const LaunchListContainerComponent = () => {
    const { data, error, loading } = useLaunchesQuery()
    const [open, setOpen] = useState(true)
    const classes = launchlist_container_error_message()

    const handleClose = (event?: React.SyntheticEvent, reason?: string) => {
        if (reason === 'clickaway') {
          return;
        }
    
        setOpen(false);
    };

    if (loading) {
        return (
            <Grid container>
               <Grid item sm={12} md={12} lg={12}>
                    <div style={{width:"100%"}}>
                        <LinearProgress/>
                    </div>
                </Grid> 
            </Grid>
        )
    }

    if (error || !data) {
        // setOpen(true)
        return(
            <Snackbar open={open}>
                <Alert onClose={handleClose} severity="error">
                  Internet connection error!
                </Alert>
            </Snackbar>
        )
    }

    console.log("Launches list fetched from container component.")
    console.log(data)

    return (
        <div>
            <Grid container>
                <Grid item sm={12} md={12} lg={12}>
                    <LaunchList data={data} />
                </Grid>
            </Grid>
        </div>
    )
}

export default LaunchListContainerComponent
