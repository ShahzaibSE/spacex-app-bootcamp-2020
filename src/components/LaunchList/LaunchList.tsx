import React,{FC} from 'react'
import Grid from "@material-ui/core/Grid"
import {AnimationWrapper} from "react-hover-animation"
// Component.
import Launch from "../Launch/Launch"
// graphql.
import {LaunchesQuery} from "../../generated/graphql"
// Assets.
import "./LaunchList.scss"

type LaunchListProps = {
    data: LaunchesQuery
}

const LaunchList: FC<LaunchListProps> = ({data}:any) => {
    console.log("Launches list sent as props")
    console.log(data)
    return (
        <div className="launch_list_container">
            <Grid container>
                    <Grid item sm={4} md={4} lg={4}>  
                        <AnimationWrapper config={{
                                                    transform:{initial:'scale(1)',onHover:'scale(1.1)'},
                                                    opacity: {initial:'1',onHover:'1'}
                                                }}>
                            <Launch/>   
                        </AnimationWrapper>    
                    </Grid>
            </Grid>
        </div>
    )
}

export default LaunchList
