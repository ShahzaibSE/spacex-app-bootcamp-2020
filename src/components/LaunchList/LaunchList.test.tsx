import React from "react";
import {shallow, mount} from "enzyme";
// Component.
import LaunchList from "./LaunchList";
import LaunchListContainerComponent from "./LaunchList.container";
// GraphQL query.
import {LaunchesQuery, useLaunchesQuery} from "../../generated/graphql";
import {ApolloProvider} from "@apollo/client";
import {client} from "../../index";

describe("<LaunchListContainerComponent> tests", ()=>{
    let container: any;
    // const {data, loading, error} = useLaunchesQuery()
    let dummy_data:any = [{flight_number:1, mission_name:"Falcon"}]

    beforeEach(()=>(
        container = shallow(<ApolloProvider client={client}><LaunchListContainerComponent/></ApolloProvider>)
    ))

    it("check if <LaunchListContainerComponent> has rendered", ()=>{
        expect(container).toEqual(true)
    })

    it("has to be at least one <div>", ()=>{
        expect(container.find("div").length).toBeGreaterThanOrEqual(1)
    })

    it("check <LaunchList> component within <LaunchListContainerComponent>",()=>{
        expect(container.containsMatchingElement(<LaunchList data={dummy_data} />)).toEqual(true)
    })
})