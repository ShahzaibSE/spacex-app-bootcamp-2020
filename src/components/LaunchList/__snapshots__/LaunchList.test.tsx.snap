// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LaunchListContainerComponent> tests check if <LaunchListContainerComponent> has rendered 1`] = `undefined`;
